resource-validator/src/test/scala/com/broadinstitute/dsp/resourceValidator/StoppedRuntimeCheckerSpec.scala

```diff
@@ -0,0 +1,179 @@
+        ): IO[Option[Instance]] = {
+          val instance = Instance.newBuilder().setStatus("STOPPED").build()
+          IO.pure(Some(instance))
+        }
+
+        override def stopInstance(project: GoogleProject, zone: ZoneName, instanceName: InstanceName)(
+          implicit ev: Ask[IO, TraceId]
+        ): IO[Operation] = if (dryRun) IO.raiseError(fail("this shouldn't be called")) else IO.pure(defaultOperation)
+      }
+
+      val dataprocService = new BaseFakeGoogleDataprocService {
+        override def getCluster(project: GoogleProject, region: RegionName, clusterName: DataprocClusterName)(
+          implicit ev: Ask[IO, TraceId]
+        ): IO[Option[Cluster]] =
+          IO.pure(Some(makeClusterWithStatus("RUNNING")))
+
+        override def getClusterInstances(project: GoogleProject,
+                                         region: RegionName,
+                                         clusterName: DataprocClusterName)(implicit ev: Ask[IO, TraceId]) =
+          IO.pure(dataprocRoleZonePreemptibilityInstances)
+
+        override def stopCluster(project: GoogleProject,
+                                 region: RegionName,
+                                 clusterName: DataprocClusterName,
+                                 metadata: Option[Map[String, String]])(
+          implicit ev: Ask[IO, TraceId]
+        ): IO[List[Operation]] =
+          if (dryRun) IO.raiseError(fail("this shouldn't be called")) else IO.pure(List(defaultOperation))
+      }
+
+      val runtimeCheckerDeps =
+        initRuntimeCheckerDeps(googleComputeService = computeService, googleDataprocService = dataprocService)
+
+      val stoppedRuntimeChecker = StoppedRuntimeChecker.iml(dbReader, runtimeCheckerDeps)
+      val res = stoppedRuntimeChecker.checkResource(runtime, dryRun)
+      res.unsafeRunSync() shouldBe None
+    }
+  }
+}
+
+object StoppedRuntimeCheckerSpec {
+  val defaultOperation = Operation.getDefaultInstance
+  val zone = ZoneName("us-central1-a")
+  val dataprocRoleZonePreemptibilityInstances = Map(
+    DataprocRoleZonePreemptibility(Master, zone, false) -> Set(InstanceName("master-instance")),
+    DataprocRoleZonePreemptibility(Worker, zone, false) -> Set(InstanceName("worker-instance-0"),
+                                                               InstanceName("worker-instance-1")),
+    DataprocRoleZonePreemptibility(SecondaryWorker, zone, true) -> Set(InstanceName("secondary-worker-instance-0"),
+                                                                       InstanceName("secondary-worker-instance-1"),
+                                                                       InstanceName("secondary-worker-instance-2"))
+  )
+
+  def makeClusterWithStatus(status: String): Cluster = {
+    val validStatus = status match {
+      case "RUNNING"  => "RUNNING"
+      case "CREATING" => "CREATING"
+      case "ERROR"    => "ERROR"
+      case "DELETED"  => "DELETING"
+      case _          => "UNRECOGNIZED"
```

Comment:
when will `_` happen?

Reply:
That's to guard against a caller passing in some other string. It wouldn't happen if the caller uses `genRuntime` but we can't know that. `genRuntime` may also change in the future.

Reply:
wouldn't that be a bug? since this is in test, I woudln't mind just throwing an exception if the input string is any invalid status (or making the function return `Option` or `Either` if you feel like).

Reply:
I don't think it would necessarily be a bug but getting an exception in such a case sounds good.